client/hudwidget.cpp

```diff
@@ -1793,14 +1795,15 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
   QString ahploss, dhploss;
 
   if (att_hp_loss > 0) {
-    ahploss = "-" + QString::number(att_hp_loss);
+    // TRANS: HP - Hit Points
+    ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
   } else {
-    ahploss = QStringLiteral("0");
+    ahploss = QString(_("%1 HP")).arg(att_hp_loss);
```

Comment:
I had a hard time understanding what the two branches of the `if` do. The `true` branch displays the HP loss if it is positive and the second branch is for no HP loss. Assuming that `att_hp_loss` is never negative, we could simplify this beast to:
```cpp
// TRANS: Unit lost %1 hit points in combat. Keep short.
ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
```
And similar for `dhploss`.

Reply:
I mostly just kept what was there and only modified to suit the issue. I wasn't looking to improve/refactor anything. I can look at this, however.

Reply:
I think we need both sides. 
```  
  if (att_hp_loss > 0) {
    // TRANS: HP - Hit Points
    ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
  } else {
    ahploss = QString(_("%1 HP")).arg(att_hp_loss);
  }
```
If HP loss is greater than 0 (meaning the unit took a hit and lost some HP), then we show a negative number in the text. We then assume in the `else` statement that there was no loss to HP and give a "positive" number that will essentially say "0 HP". I think this is more about the text setup. We don't want to show "-0 HP".

Reply:
I kind of assumed that `-0 HP` was fine (would be useful to distinguish lost HP from remaining HP). Otherwise it can be `ahploss = QString(_("%1 HP")).arg(-att_hp_loss)`